src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { loadAgencies } from '../../redux/modules/agency'
+
+class AgencyList extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      loading: false,
+      agencies: [],
+      saved: null
+    }
+
+  }
+
+  componentDidMount() {
+    this.setState({
+      loading: true
+    })
+    this.props.loadAgencies()
+      .then(r => {
+        if (r.status === 200) {
+          return r.json()
+        }
+        this.setState({
+          loading: false
+        })
+        return Promise.reject()
+      })
+      .then(r => {
+        this.setState({
+          loading: false,
+          agencies: r
```

Comment:
If the promise is rejected above, then `r` here will not likely be in an expected format. If the `agencies` state change is moved to the previous `.then()` inside the success condition, then this `.then()` can be removed.